torch/nn/modules/transformer.py

```diff
@@ -126,14 +127,14 @@ def forward(self, src: Tensor, tgt: Tensor, src_mask: Optional[Tensor] = None, t
         r"""Take in and process masked source/target sequences.
 
         Args:
-            src: the sequence to the encoder (required).
-            tgt: the sequence to the decoder (required).
-            src_mask: the additive mask for the src sequence (optional).
-            tgt_mask: the additive mask for the tgt sequence (optional).
-            memory_mask: the additive mask for the encoder output (optional).
-            src_key_padding_mask: the Tensor mask for src keys per batch (optional).
```

Comment:
how come we remove the optional keyword in the new version?

Reply:
The typing annotations (not here but auto-generated by sphinx) already reflect which args are optional/required

e.g. 
<img width="814" alt="Screenshot 2023-11-29 at 3 47 01 PM" src="https://github.com/pytorch/pytorch/assets/35276741/0526026f-61b8-4674-978c-e668fa2cc1b9">